### PR TITLE
Fix WithDockerfileTests container start failures.

### LINKED ISSUE #4792: Container start failures due to missing executables in container don't show in the dashboard.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If I start a container and the EXEC command that I have in the Dockerfile doesn't find an executable the container will fail to launch and the underlying error from the container runtime won't be shown in the portal.

### Expected Behavior

_No response_

### Steps To Reproduce

Use this Dockerfile

```dockerfile
FROM ubuntu
RUN echo hello-world
CMD [ "doestnotexist", "42" ]
```

And this apphost:

```csharp
var builder = DistributedApplication.CreateBuilder(args);

builder.AddDockerfile("mycontainer", ".");

builder.Build().Run();
```

When you run it you'll get this in your console logs for the AppHost:

```
fail: Aspire.Hosting.Dcp.dcpctrl.ContainerReconciler[0]
      could not start the container	{"Container": {"name":"ddd-rttkcvtp"}, "Reconciliation": 5, "ContainerID": "2da792a7c27aefff14ba96eb5bd75c4d404d25783cce3a5c008a81e6ccf6d1cf", "error": "docker command 'StartContainers' returned with non-zero exit code 1: command output: Stdout: '' Stderr: 'Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: \"doestnotexist\": executable file not found in $PATH: unknown\nError: failed to start containers: 2da792a7c27aefff14ba96eb5bd75c4d404d25783cce3a5c008a81e6ccf6d1cf\n'"}
```

But the dashboard will only show the output for the image pull for the base image.

### Exceptions (if any)

_No response_

### .NET Version info

.NET SDK:
 Version:           8.0.105
 Commit:            eae90abaaf
 Workload version:  8.0.100-manifests.b4a94546

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/lib/dotnet/sdk/8.0.105/

.NET workloads installed:
 Workload version: 8.0.100-manifests.b4a94546
 [aspire]
   Installation Source: SDK 8.0.100
   Manifest Version:    8.1.0-preview.1.24353.12/8.0.100
   Manifest Path:       /home/mitch/.dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.aspire/8.1.0-preview.1.24353.12/WorkloadManifest.json
   Install Type:        FileBased

Host:
  Version:      8.0.5
  Architecture: x64
  Commit:       087e15321b

.NET SDKs installed:
  7.0.119 [/usr/lib/dotnet/sdk]
  8.0.105 [/usr/lib/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.19 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.5 [/usr/lib/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.19 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.5 [/usr/lib/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR fixes #4792.

The issue was that the team that owns that image on MCR significantly changed the way it works (it just runs an executable rather than a web server that can service static content by the looks of it). To address this I changed to using the Mariner-based Nginx image also hosted on mcr.microsoft.com (because supply chain issues).

This resulted in a few necessary changes to the Dockerfile setups. Should all be unblocked now.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4832)

### CODE REVIEW

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
> This PR fixes #4792.

I think you meant https://github.com/dotnet/aspire/issues/4828 .

File: tests/Aspire.Hosting.Containers.Tests/WithDockerfileTests.cs [hunk 1]
```diff
@@ -822,28 +822,33 @@ private static async Task WaitForResourceAsync(DistributedApplication app, strin
...
-        COPY --from=builder /app/static/cachebuster.txt /app/static
-        COPY --from=builder /app/static/aspire.html /app/static
-        RUN --mount=type=secret,id=FILE_SECRET cp /run/secrets/FILE_SECRET /app/static/FILE_SECRET.txt
-        RUN --mount=type=secret,id=ENV_SECRET cp /run/secrets/ENV_SECRET /app/static/ENV_SECRET.txt
+        RUN mkdir -p /usr/share/nginx/html
+        COPY --from=builder /usr/share/nginx/html/cachebuster.txt /usr/share/nginx/html
+        COPY --from=builder /usr/share/nginx/html/aspire.html /usr/share/nginx/html
+        RUN --mount=type=secret,id=FILE_SECRET cp /run/secrets/FILE_SECRET /usr/share/nginx/html/FILE_SECRET.txt
+        RUN --mount=type=secret,id=ENV_SECRET cp /run/secrets/ENV_SECRET /usr/share/nginx/html/ENV_SECRET.txt
+        RUN chmod -R 777 /usr/share/nginx/html
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Containers.Tests/WithDockerfileTests.cs:852]: Why is this necessary?

Comment (mitchdenny) [reply tests/Aspire.Hosting.Containers.Tests/WithDockerfileTests.cs:852]: Because in the old container the directory already existed. In the new container it didn't so it inherited the permissions from the parent directory so I just loosened off the permissions so that the user that nginx is running as can read the content to serve it.

Comment (eerhardt) [reply tests/Aspire.Hosting.Containers.Tests/WithDockerfileTests.cs:852]: Why isn't it needed in the `HelloWorldDockerfile` version?
